### PR TITLE
Comment out engine gcs filesystem in config Closes #705

### LINKED ISSUE #705: "Getting started" with Cromwell includes ADC exception
For users just getting started with Cromwell, the documentation here:

https://github.com/broadinstitute/cromwell#getting-started-with-wdl

Points the user here:

https://github.com/broadinstitute/wdl#getting-started-with-wdl

and when you run your first workflow, you'll see in the output:

```
[2016-04-14 16:21:12,82] [warn] Failed to get application default credentials
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at com.google.api.client.googleapis.auth.oauth2.DefaultCredentialProvider.getDefaultCredential(DefaultCredentialProvider.java:93)
	at com.google.api.client.googleapis.auth.oauth2.GoogleCredential.getApplicationDefault(GoogleCredential.java:213)
	at com.google.api.client.googleapis.auth.oauth2.GoogleCredential.getApplicationDefault(GoogleCredential.java:191)
	at cromwell.util.google.GoogleCredentialFactory.forApplicationDefaultCredentials(GoogleCredentialFactory.scala:125)
	at cromwell.util.google.GoogleCredentialFactory.fromCromwellAuthScheme$lzycompute(GoogleCredentialFactory.scala:64)
	at cromwell.util.google.GoogleCredentialFactory.fromCromwellAuthScheme(GoogleCredentialFactory.scala:61)
	at cromwell.engine.backend.io.filesystem.gcs.StorageFactory$$anonfun$cromwellAuthenticated$1.apply(StorageFactory.scala:20)
	at cromwell.engine.backend.io.filesystem.gcs.StorageFactory$$anonfun$cromwellAuthenticated$1.apply(StorageFactory.scala:20)
	at scala.util.Try$.apply(Try.scala:192)
	at cromwell.engine.backend.io.filesystem.gcs.StorageFactory$.cromwellAuthenticated$lzycompute(StorageFactory.scala:20)
	at cromwell.engine.backend.io.filesystem.gcs.StorageFactory$.cromwellAuthenticated(StorageFactory.scala:18)
	at cromwell.engine.backend.local.LocalBackend$$anonfun$14.apply(LocalBackend.scala:239)
	at cromwell.engine.backend.local.LocalBackend$$anonfun$14.apply(LocalBackend.scala:239)
	at scala.util.Try.orElse(Try.scala:84)
	at cromwell.engine.backend.local.LocalBackend.fileSystems(LocalBackend.scala:239)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor$$anonfun$cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor$1$1.apply(MaterializeWorkflowDescriptorActor.scala:89)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor$$anonfun$cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor$1$1.apply(MaterializeWorkflowDescriptorActor.scala:86)
	at scalaz.ValidationFlatMap.flatMap(Validation.scala:433)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor.cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor$1(MaterializeWorkflowDescriptorActor.scala:86)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor$$anonfun$cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor$2$$anonfun$apply$8.apply(MaterializeWorkflowDescriptorActor.scala:110)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor$$anonfun$cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor$2$$anonfun$apply$8.apply(MaterializeWorkflowDescriptorActor.scala:109)
	at scalaz.ValidationFlatMap.flatMap(Validation.scala:433)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor$$anonfun$cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor$2.apply(MaterializeWorkflowDescriptorActor.scala:109)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor$$anonfun$cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor$2.apply(MaterializeWorkflowDescriptorActor.scala:103)
	at scalaz.ValidationFlatMap.flatMap(Validation.scala:433)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor.cromwell$engine$workflow$MaterializeWorkflowDescriptorActor$$buildWorkflowDescriptor(MaterializeWorkflowDescriptorActor.scala:103)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor$$anonfun$receive$1.applyOrElse(MaterializeWorkflowDescriptorActor.scala:66)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:467)
	at cromwell.engine.workflow.MaterializeWorkflowDescriptorActor.aroundReceive(MaterializeWorkflowDescriptorActor.scala:59)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
	at akka.actor.ActorCell.invoke(ActorCell.scala:487)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
	at akka.dispatch.Mailbox.run(Mailbox.scala:220)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

To a novice user it is not clear whether that this can safely be ignored. This is a problem in particular if this first use actually fails for some other reason. The user will spend time trying to figure out if the problem is caused by a credential issue.

I tried to see if I could easily suppress this by setting up "backends" to only include "local". I pulled down the [application.conf](https://github.com/broadinstitute/cromwell/blob/9f759a54a0b8873f1338f36391f985477d83475a/engine/src/main/resources/application.conf) which sets:

```
backend {
  // Either "jes", "local", or "sge" (case insensitive)
  defaultBackend = "local"
  // List of backends which this Cromwell supports. Be sure to include the defaultBackend!
  backendsAllowed = [ "local" ]
```

and then passed the file as:

`$ java -Dconfig.file=application.conf -jar cromwell.jar run hello.wdl hello.json
`
But the exception and warning were still raised.

Should the ADC check be occurring even when backendsAllowed does not include JES?

Thanks.

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by mcovarr** [comment]
whoops :+1:

[Approved with PullApprove](https://pullapprove.com/broadinstitute/cromwell/pull-request/1135/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/cromwell)

**Comment by geoffjentry** [comment]
👍

**Comment by cjllanwarne** [comment]
Maybe a comment to say when it *is* appropriate to have this in?

If FireCloud (which is using JES) doesn't need this, who does...?

File: core/src/main/resources/application.conf [hunk 1]
```diff
@@ -115,13 +115,13 @@ google {
...
 }
 
-engine {
-  filesystems {
-    gcs {
-      auth = "application-default"
-    }
-  }
-}
+//engine {
```
Comment (cjllanwarne) [inline core/src/main/resources/application.conf]: Maybe leave the `engine` block in, and just have the `filesystems` commented out?

**Comment by Horneth** [comment]
@cjllanwarne Firecloud needs it, this is to avoid cromwell to always require a google auth (even when running only local backend).

**Comment by kcibul** [comment]
really clear comments in the config file, nice

**Pull request merged**